### DISCUSSION TITLE
Get original tweet ID from retweet

### DISCUSSION THREAD

**Discussion by nmduarteus** [plhery/node-twitter-api-v2#266] [Q&A]
While looking at tweets (that can be retweets) I would like to see/know the original tweet ID. is that possible?

Example:
The Tweet text will show something like `"text": "RT @ABC: Thank you!"` - One could assume that if the tweet text starts with `RT @handler` then it means it's a retweet. But is there any way of retrieving which tweet was retweeted?

Thank you!

**Comment by alkihis** [comment]
Hi,
You can get RT data using `expansions` in request, then read them using `.includes` in response.
Lib also provides an helper to simplify usage and findings in those includes:

```ts
const originalTweet = await client.v2.singleTweet('20', { expansions: ['referenced_tweets.id'], 'tweet.fields': ['referenced_tweets'] })

// Without includes helper,
const rtID = originalTweet.data.referenced_tweets?.find(r => r.type === 'retweeted')?.id

if (rtID) {
  console.log('RT ID', rtID)
} else {
  console.log('Not a RT')
}

// With includes helper
const includes = new TwitterV2IncludesHelper(originalTweet)
const retweet = includes.retweet(originalTweet.data)

if (retweet) {
  console.log('This is a RT, data:', retweet) // id is in {retweet.id}
} else {
  console.log('Not a RT, tweet data:', originalTweet.data)
}
```
